src/ring_theory/roots_of_unity.lean

```diff
@@ -907,9 +907,8 @@ begin
   replace habs := lt_of_lt_of_le (enat.coe_lt_coe.2 one_lt_two)
     (multiplicity.le_multiplicity_of_pow_dvd (dvd_trans habs prod)),
   have hfree : squarefree (X ^ n - 1 : polynomial (zmod p)),
-  { refine squarefree_X_pow_sub_C 1 _ one_ne_zero,
-    by_contra hzero,
-    exact hdiv ((zmod.nat_coe_zmod_eq_zero_iff_dvd n p).1 hzero) },
+  { exact (separable_X_pow_sub_C 1
+          (λ h, hdiv $ (zmod.nat_coe_zmod_eq_zero_iff_dvd n p).1 h) one_ne_zero).squarefree },
```

Comment:
There is some weird issues that happen if you leave in the original proof but with `.squarefree` instead; I'm happy to revert this lemma deletion if you guys think this is worth it.